homeassistant/components/myuplink/binary_sensor.py

```diff
@@ -98,3 +136,73 @@ def is_on(self) -> bool:
         """Binary sensor state value."""
         device_point = self.coordinator.data.points[self.device_id][self.point_id]
         return int(device_point.value) != 0
+
+    @property
+    def available(self) -> bool:
+        """Return device data availability."""
+        return super().available and (
+            self.coordinator.data.devices[self.device_id].connectionState
+            == DeviceConnectionState.Connected
+        )
+
+
+class MyUplinkDeviceBinarySensor(MyUplinkEntity, BinarySensorEntity):
+    """Representation of a myUplink device bound binary sensor."""
+
+    def __init__(
+        self,
+        coordinator: MyUplinkDataCoordinator,
+        device_id: str,
+        entity_description: BinarySensorEntityDescription | None,
+        unique_id_suffix: str,
+    ) -> None:
+        """Initialize the binary_sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            device_id=device_id,
+            unique_id_suffix=unique_id_suffix,
+        )
+
+        if entity_description is not None:
+            self.entity_description = entity_description
+
+    @property
+    def is_on(self) -> bool:
+        """Binary sensor state value."""
+        return (
+            self.coordinator.data.devices[self.device_id].connectionState
+            == DeviceConnectionState.Connected
+        )
+
+
+class MyUplinkSystemBinarySensor(MyUplinkSystemEntity, BinarySensorEntity):
+    """Representation of a myUplink system bound binary sensor."""
+
+    def __init__(
+        self,
+        coordinator: MyUplinkDataCoordinator,
+        system_id: str,
+        device_id: str,
+        entity_description: BinarySensorEntityDescription | None,
+        unique_id_suffix: str,
+    ) -> None:
+        """Initialize the binary_sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            system_id=system_id,
+            device_id=device_id,
+            unique_id_suffix=unique_id_suffix,
+        )
+
+        if entity_description is not None:
+            self.entity_description = entity_description
+
+    @property
+    def is_on(self) -> bool | None:
+        """Binary sensor state value."""
+        retval = None
+        for system in self.coordinator.data.systems:
+            if system.id == self.system_id:
```

Comment:
It would be nice if this was a dict so we didn't have to look the system up every time.

This could then be its separate property for easy access

Reply:
I put the dict comprehension here for now. It could be moved to the DataUpdateCoordinator, but this is currently the only place we access one specific system in the integration. I would prefer to keep it this way for now. The conversion of Systems to a dict would have implications in many places in the integration and I think it is better to do that in a separate PR.

Reply:
Agreed. But since you only use this in one place now (which I didn't know) I think I like the for loop more as that would be more efficient at times

Reply:
Reverting!